src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -775,13 +762,14 @@ private static bool IsByValueClass(TypeDesc type)
             return type.IsValueType && !type.IsPrimitive && !type.IsEnum;
         }
 
-        private static LayoutInt ComputeBytesUsedInParentType(DefType type)
+        public LayoutInt CalculateFieldBaseOffset(TypeDesc type)
```

Comment:
This API will crash with an InvalidCast if e.g. an array is passed. It would be better to ensure valid inputs at compile time.

```suggestion
        public LayoutInt CalculateFieldBaseOffset(MetdataType type)
```

Reply:
Good point, thanks, fixed in 8th commit.